Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -360,11 +359,11 @@ private func createResolvedPackages(
                     // dependency to share the same name. We don't check this in manifest loading for root-packages so
                     // we can provide a more detailed diagnostic here.
                     let referencedPackageURL = mirrors.effectiveURL(forURL: product.packageBuilder.package.manifest.url)
-                    let referencedPackageIdentity = PackageReference.computeIdentity(packageURL: referencedPackageURL)
+                    let referencedPackageIdentity = PackageIdentity(referencedPackageURL)
                     let packageDependency = packageBuilder.package.manifest.dependencies.first { package in
                         let packageURL = mirrors.effectiveURL(forURL: package.url)
-                        let packageIdentity = PackageReference.computeIdentity(packageURL: packageURL)
-                        return packageIdentity == referencedPackageIdentity
+                        let packageIdentity = PackageIdentity(packageURL)
+                        return packageIdentity.computedName == referencedPackageIdentity.computedName
```

Comment:
comparison on computedName instead of PackageIdentity intentional?

Reply:
Yes. Changing to a direct comparison of `PackageIdentity` values would change the semantics of package identity.